app/src/main/java/invoice_automation/module/QuickBooksModule.java

```diff
@@ -83,8 +84,34 @@ public List<Customer> getAllCustomers() {
      * @return A copy of the Customer updated or created
      */
     public Customer updateCustomerFromSchool(@NonNull School school) {
-        // TODO: IA-7
-        return null;
+        // Get customer associated with the school, if it exists
+        Customer schoolCustomer = new Customer();
+        List<Customer> customers = getAllCustomers();
+        for (Customer c: customers) {
+            if (c.getDisplayName().equals(school.getSchoolName())) {
+                schoolCustomer.setId(c.getId());
+                schoolCustomer.setSyncToken(c.getSyncToken());
+                break;
+            }
+        }
+        // Convert the school object to a customer object
+        Customer newCustomer = QuickBooksUtils.getCustomerFromSchool(school);
+        if (schoolCustomer.getId() != null) {
+            newCustomer.setId(schoolCustomer.getId());
+            newCustomer.setSyncToken(schoolCustomer.getSyncToken());
+        }
+        // Add newCustomer or update the existing customer
+        try {
+            if (schoolCustomer.getId() == null) {
+                dataService.add(newCustomer);
+            } else {
+                newCustomer.setSparse(true);
+                dataService.update(newCustomer);
+            }
+        } catch (FMSException e) {
+            throw new QuickBooksException("Exception updating customer", e);
```

Comment:
Having different error messages for when adding fails vs when updating fails will make debugging easier